### PR TITLE
[REVIEW] Deprecate `inplace` parameters in categorical methods

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
 To get ready for pandas-2.0 compatibility, this PR deprecates `inplace` in the following APIs:

- [x] `as_ordered`
- [x] `as_unordered`
- [x] `add_categories`
- [x] `remove_categories`
- [x] `set_categories`
- [x] `reorder_categories`

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/column/categorical.py [hunk 1]
```diff
@@ -423,6 +458,15 @@ def remove_categories(
         dtype: category
         Categories (2, int64): [1, 2]
         """
+        if inplace:
+            warnings.warn(
+                "The `inplace` parameter in cudf.Series.cat.reorder_categories"
+                " is deprecated and will be removed in a future version of "
```
Comment (brandon-b-miller) [inline python/cudf/cudf/core/column/categorical.py:464]: nit: let's be consistent about putting the space at the end or beginning of the line.

File: python/cudf/cudf/core/column/categorical.py [hunk 2]
```diff
@@ -141,6 +141,13 @@ def as_ordered(self, inplace: bool = False) -> Optional[SeriesOrIndex]:
             or return a copy of this categorical with
             added categories.
 
+            .. deprecated:: 23.02
```
Comment (brandon-b-miller) [inline python/cudf/cudf/core/column/categorical.py:144]: I see `23.02` here but `23.04` below, is this as intended?

Comment (galipremsagar) [reply python/cudf/cudf/core/column/categorical.py:144]: Yup, this is intentional. `inplace` here was deprecated in `23.02`. This is just updating the docstring.

File: python/cudf/cudf/core/column/categorical.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/column/categorical.py]: Done 👍

**Comment by brandon-b-miller** [comment]
AFAICT this is just adding deprecation warnings, should this be `non-breaking`?

**Review by brandon-b-miller — APPROVED** [review]

**Pull request merged**